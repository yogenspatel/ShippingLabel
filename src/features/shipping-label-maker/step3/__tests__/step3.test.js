/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import ShippingLabelStep3 from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Step 3 Component renders correctly", () => {
  it("has rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ShippingLabelStep3 />, div);
  });
  it("has called snapshot with ShippingLabel component", () => {
    const tree = shallow(<ShippingLabelStep3 />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("has called componentDidMount method", () => {
    const spy = jest.spyOn(ShippingLabelStep3.prototype, 'componentDidMount');
    const props = {
        setShippingData: {},
        getShippingData: jest.fn()
    }
    const wrapper = shallow(<ShippingLabelStep3 {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
    expect(props.getShippingData).toBeCalled();
    spy.mockClear();
  });
  it("has retrieved data on componentDidMount method", () => {
    const shippingData = {
        weight: {
            weight: 5
        }
    }
    const spy = jest.spyOn(ShippingLabelStep3.prototype, 'componentDidMount');
    const props = {
        setShippingData: shippingData,
        getShippingData: jest.fn()
    }
    const wrapper = shallow(<ShippingLabelStep3 {...props} />);
    wrapper.instance().componentDidMount();
    expect(wrapper.state('weight')).toEqual(shippingData.weight.weight);
    spy.mockClear();
  });
});
