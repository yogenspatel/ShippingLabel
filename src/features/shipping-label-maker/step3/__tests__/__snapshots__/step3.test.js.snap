// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step 3 Component renders correctly has called snapshot with ShippingLabel component 1`] = `
<div
  className="card"
>
  <div
    className="h4 card-header"
  >
    Enter the Package Weight
  </div>
  <div
    className="card-body"
  >
    <RenderFormField
      context={
        ShippingLabelStep3 {
          "context": Object {},
          "key": "weight",
          "props": Object {
            "getShippingData": [Function],
            "setShippingData": Object {},
          },
          "refs": Object {},
          "setState": [Function],
          "state": Object {
            "errorObj": Object {
              "weight": true,
            },
            "weight": 0,
          },
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useState": [Function],
              },
              "_element": <ShippingLabelStep3
                getShippingData={[Function]}
                setShippingData={Object {}}
              />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <div
                className="card"
              >
                <div
                  className="h4 card-header"
                >
                  Enter the Package Weight
                </div>
                <div
                  className="card-body"
                >
                  <RenderFormField
                    context={[Circular]}
                    fieldName="weight"
                    onChange={[Function]}
                    placeHolder="Enter Weight"
                    type="number"
                  />
                </div>
              </div>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
      fieldName="weight"
      onChange={[Function]}
      placeHolder="Enter Weight"
      type="number"
    />
  </div>
</div>
`;
